### PR TITLE
Fix context data with non-String values

### PR DESCRIPTION
**Pull request by waldeinburg**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/mp911de/logstash-gelf/blob/master/.github/CONTRIBUTING.md).
- [x] You use the code formatters provided [here](https://github.com/mp911de/logstash-gelf/blob/master/as7formatter.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

File: src/main/java/biz/paluch/logging/gelf/log4j2/Log4j2LogEvent.java [hunk 1]
```diff
@@ -172,11 +173,19 @@ private String getValue(MdcMessageField field) {
     @Override
     public String getMdcValue(String mdcName) {
         ReadOnlyStringMap contextData = logEvent.getContextData();
-        if (null != contextData && contextData.containsKey(mdcName)) {
-            return contextData.getValue(mdcName);
+        if (null == contextData) {
```
Comment (mp911de) [inline src/main/java/biz/paluch/logging/gelf/log4j2/Log4j2LogEvent.java:176]: How about converting the value ourselves to `String`? Creating another map imposes quite significant allocations?

**Comment by waldeinburg** [comment]
Ouch, you're right! I actually used `String.valueOf` initially, but I changed it to this bad solution when tests failed and I didn't think about that `null` returned `"null"` because I was just wrapping `getValue`.

**Comment by waldeinburg** [comment]
I just amended the commit and force-pushed. No need to have the old solution in the history.

**Comment by waldeinburg** [comment]
Because I check for null now I could of course just use `value.toString` instead of `String.valueOf`. Do you prefer the other?

**Comment by waldeinburg** [comment]
Never mind, I changed it. `String.valueOf` is just `obj == null) ? "null" : obj.toString()` so it's silly to use it.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Pull request closed without merging**